include/aws/common/common.h

```diff
@@ -50,20 +50,24 @@
 otherwise they are checked
 */
 #ifdef CBMC
-#define AWS_PRECONDITION(cond) __CPROVER_assume(cond)
-#define AWS_POSTCONDITION(cond) assert(cond)
-#define AWS_MEM_IS_READABLE(base, len) __CPROVER_r_ok((base), (len))
-#define AWS_MEM_IS_WRITABLE(base, len) __CPROVER_w_ok((base), (len))
+#    define AWS_PRECONDITION(cond) __CPROVER_assume(cond)
+#    define AWS_MEM_IS_READABLE(base, len) __CPROVER_r_ok((base), (len))
+#    define AWS_MEM_IS_WRITABLE(base, len) __CPROVER_w_ok((base), (len))
 #else
-#define AWS_PRECONDITION(cond) assert(cond)
-#define AWS_POSTCONDITION(cond) assert(cond)
+#    define AWS_PRECONDITION(cond) assert(cond)
 /* the C runtime does not give a way to check these properties,
  * but we can at least check that the pointer is valid */
-#define AWS_MEM_IS_READABLE(base, len) (((len) == 0) || (base))
-#define AWS_MEM_IS_WRITABLE(base, len) (((len) == 0) || (base))
+#    define AWS_MEM_IS_READABLE(base, len) (((len) == 0) || (base))
+#    define AWS_MEM_IS_WRITABLE(base, len) (((len) == 0) || (base))
 #endif
 
-
+/**
+ * Define a function contract to encapsulate the check of a postcondition with
+ * the return value of a function. This keep the assert in the same line for debuggability.
+ * If NDEBUG is on, this will have 0 runtime cost.
+ */
+#define AWS_RETURN_VALUE_WITH_POSTCONDITION(cond, val) (assert(cond), (val))
+#define AWS_RETURN_SUCCESS_WITH_POSTCONDITION(cond) AWS_RETURN_VALUE_WITH_POSTCONDITION((cond), AWS_OP_SUCCESS)
```

Comment:
What macro should we use for void functions?

Reply:
`AWS_POSTCONDITION` only? As we currently have